bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/GenericCommunicator.java

```diff
@@ -197,10 +205,13 @@ private void step7FinalInitializationRequest(byte[] initializationMessage)
             logger.warn("Logon to panel failure.");
             logoutSequence();
         }
-        Thread.sleep(300);
-        // TODO check why after a short sleep, a 37 bytes packet is received after logon
-        // ! ! !
-        receivePacket();
+
+        try {
+            scheduler.schedule(this::receivePacket, 300, TimeUnit.MILLISECONDS).get();

```

Comment:
Because of the `get  it does exactly the same, but why do you / would you need the get?

Reply:
I'd like not to move forward unless I have received this "ghost" 37 byte packet which I'm not aware what exactly does. :) Otherwise this packet will get received together with next call and will mess up the normal communication. Since this is relevant only for the login sequence and usually happens only during initialization/reset I believe it's not a big deal even with Thread.sleep().

Reply:
Sorry, but I think that is really error prone and not good. Because this method is indirectly called from the constructor so you are blocking a thread, still effectively almost the same as the sleep which we had imho, inside a constructor. Cant you just implement some logic to ignore this ghost packet whenever it arrives? Or just ignore the first packet after a login. 

@openhab/2-x-add-ons-maintainers do you have any great ideas to resolve this (quickly)